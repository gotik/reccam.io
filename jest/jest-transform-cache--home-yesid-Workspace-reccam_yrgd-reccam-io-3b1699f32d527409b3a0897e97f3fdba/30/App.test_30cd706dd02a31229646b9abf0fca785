/* istanbul ignore next */'use strict';var _regenerator = require('/home/yesid/Workspace/reccam_yrgd/reccam.io/node_modules/react-scripts/node_modules/babel-runtime/regenerator');var _regenerator2 = _interopRequireDefault(_regenerator);var /* istanbul ignore next */_react = require('react'); /* istanbul ignore next */var _react2 = _interopRequireDefault(_react);
var /* istanbul ignore next */_reactTestRenderer = require('react-test-renderer'); /* istanbul ignore next */var _reactTestRenderer2 = _interopRequireDefault(_reactTestRenderer);
var /* istanbul ignore next */_App = require('../src/App'); /* istanbul ignore next */var _App2 = _interopRequireDefault(_App);
var /* istanbul ignore next */_Deferred = require('./utils/Deferred'); /* istanbul ignore next */var _Deferred2 = _interopRequireDefault(_Deferred);
var /* istanbul ignore next */_NavigatorUserMediaError = require('./utils/NavigatorUserMediaError'); /* istanbul ignore next */var _NavigatorUserMediaError2 = _interopRequireDefault(_NavigatorUserMediaError);
var /* istanbul ignore next */_MediaStream = require('./utils/MediaStream'); /* istanbul ignore next */var _MediaStream2 = _interopRequireDefault(_MediaStream);function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}function _asyncToGenerator(fn) {return function () {var gen = fn.apply(this, arguments);return new Promise(function (resolve, reject) {function step(key, arg) {try {var info = gen[key](arg);var value = info.value;} catch (error) {reject(error);return;}if (info.done) {resolve(value);} else {return Promise.resolve(value).then(function (value) {return step("next", value);}, function (err) {return step("throw", err);});}}return step("next");});};}

var videoURL = 'http://localhost/e75dbc24-dd2c-4157-bf0e-f0a7773e98c8';
function createObjectURL(stream) {
  var className = Object.getPrototypeOf(stream).constructor.name;
  if (className === 'Blob' || className === 'MediaStream') {
    return videoURL;
  }
}

describe('App', function () {
  var deferred = /* istanbul ignore next */void 0;
  var component = /* istanbul ignore next */void 0;

  beforeAll(function () {
    URL.createObjectURL = createObjectURL;
  });

  beforeEach(function () {
    deferred = new /* istanbul ignore next */_Deferred2.default();
    navigator.mediaDevices = { getUserMedia: function /* istanbul ignore next */getUserMedia(constraints) /* istanbul ignore next */{return deferred.promise;} };
    component = /* istanbul ignore next */_reactTestRenderer2.default.create( /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */_App2.default, /* istanbul ignore next */null));
  });

  it('renders null while loading', function () {
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('renders the video URL from the stream', /* istanbul ignore next */_asyncToGenerator(_regenerator2.default.mark(function /* istanbul ignore next */_callee() /* istanbul ignore next */{var stream;return _regenerator2.default.wrap(function _callee$(_context) {while (1) {switch (_context.prev = _context.next) {case 0:
            stream = new /* istanbul ignore next */_MediaStream2.default();
            deferred.resolve(stream);_context.next = 4;return (

              deferred.promise);case 4:
            expect(component.toJSON()).toMatchSnapshot();case 5:case 'end':return _context.stop();}}}, _callee, undefined);})));


  it('renders null when media access is denied', /* istanbul ignore next */_asyncToGenerator(_regenerator2.default.mark(function /* istanbul ignore next */_callee2() /* istanbul ignore next */{var error;return _regenerator2.default.wrap(function _callee2$(_context2) {while (1) {switch (_context2.prev = _context2.next) {case 0:
            error = new /* istanbul ignore next */_NavigatorUserMediaError2.default();
            deferred.reject(error);_context2.prev = 2;_context2.next = 5;return (


              deferred.promise);case 5:_context2.next = 10;break;case 7:_context2.prev = 7;_context2.t0 = _context2['catch'](2);

            expect(component.toJSON()).toMatchSnapshot();case 10:case 'end':return _context2.stop();}}}, _callee2, undefined, [[2, 7]]);})));


});