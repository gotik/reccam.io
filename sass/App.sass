@import compass/reset

html,
body
  height: 100%

// React specific elements
#root,
[data-reactroot]
  height: inherit

[data-reactroot]
  display: flex
  justify-content: center

.fullscreen
  height: auto
  left: 50%
  min-height: 100%
  min-width: 100%
  position: fixed
  top: 50%
  transform: translate(-50%, -50%)
  width: auto
  z-index: -1

.buttons-container
  align-items: center
  align-self: flex-end
  display: flex
  justify-content: space-around
  padding-bottom: 10px
  width: 100%
  max-width: 640px

[class*="button-"]
  background-color: transparent
  border: 2px solid white
  color: white
  outline: none
  padding: 8px

  &[disabled]
    color: rgba(white, .6)
    background-color: rgba(white, .2)
    border-color: rgba(white, .6)

@mixin circle($size)
  border-radius: $size/2
  height: $size
  min-height: $size
  min-width: $size
  width: $size

.button-medium
  @include circle(60px)
  &.fa
    font-size: 26px

.button-large
  @include circle(80px)
  &.fa
    font-size: 42px