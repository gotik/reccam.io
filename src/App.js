import React, { Component } from 'react';
import mediaConstraints from './config/media-constraints';
import './App.css';

class App extends Component {
  state = { isLoading: true }

  componentWillMount() {
    navigator.mediaDevices.getUserMedia(mediaConstraints)
      .then(this.onCameraStream)
      .catch(this.onCameraError)
  }

  onCameraStream = (stream) => {
    const videoURL = URL.createObjectURL(stream);
    this.setState({ videoURL, isLoading: false });
  }

  onCameraError = ({ message: error }) => {
    this.setState({ error });
  }

  render() {
    if (this.state.isLoading) return null;

    return <video src={this.state.videoURL} muted loop autoPlay />;
  }
}

export default App;
