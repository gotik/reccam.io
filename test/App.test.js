import React from 'react';
import renderer from 'react-test-renderer';
import App from '../src/App';
import Deferred from './utils/Deferred';

describe('App', () => {
  const getUserMediaDeferred = new Deferred();

  beforeEach(() => {
    URL.createObjectURL = stream => 'http://some_url';
    navigator.mediaDevices = { getUserMedia: constraints => getUserMediaDeferred.promise };
  });

  it('renders null while loading', () => {
    const component = renderer.create(<App />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('renders the video URL from the stream', async () => {
    const component = renderer.create(<App />);

    getUserMediaDeferred.resolve({});
    await getUserMediaDeferred.promise;

    expect(component.toJSON()).toMatchSnapshot();
  });
});
