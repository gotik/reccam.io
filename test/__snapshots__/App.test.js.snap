exports[`App renders null when media access is denied 1`] = `null`;

exports[`App renders null while loading 1`] = `null`;

exports[`App renders the video URL from the stream 1`] = `
<video
  autoPlay={true}
  loop={true}
  muted={true}
  src="http://localhost/e75dbc24-dd2c-4157-bf0e-f0a7773e98c8" />
`;
